plugins/sh/src/com/intellij/sh/lexer/Sh.flex

```diff
@@ -29,6 +29,7 @@ import com.intellij.openapi.util.text.StringUtil;
   private boolean isQuoteOpen;
   private String heredocMarker;
   private boolean heredocWithWhiteSpaceIgnore;
+  private int herestringStart = -1;
```

Comment:
I renamed it to `herestringStartPosition` to understand clearly what it stores.
I also added the reset of this variable at the `onReset()` method.